libraries/stdlib/genosl/include/mx_funcs.h

```diff
@@ -5,49 +5,25 @@
 // http://www.materialx.org/
 
 #pragma once
+
 #include "color4.h"
 #include "vector2.h"
 #include "vector4.h"
 #include "matrix33.h"
 
+//
+// Support functions for OSL implementations of the MaterialX nodes.
+//
+
+float mx_ternary(int expr, float v1, float v2) { if (expr) return v1; else return v2; }
```

Comment:
N.B. You can say `expr ? v1 : v2` in OSL.

Reply:
@lgritz I took that approach at first, but it looks like OSL's built-in ternary operator doesn't yet handle values that are structures, which is the case when `v1` and `v2` are of type `color4`, `vector4`, `vector2`, or `matrix33`.  Let me know if I should write this up as an issue on the OSL GitHub, or if perhaps I'm missing a detail that would allow me to use this approach across all types.

Reply:
OH. Hmmm. Yeah, please file an issue. You can 'return' a struct, and you can assign a struct, so probably there's just some little missing bit of logic that prevents a struct from being the operands to a ternary operator.

Of course, 50/50 chance I simply don't remember that I looked at this ages ago and bailed when it turned out to be disproportionately difficult. Sometimes you turn over a rock, and when you see what's under there, you just put the rock back and slowly back away. :-)